report.md

```diff
@@ -61,6 +61,32 @@ Each of the following requirements will be linked to new tests, since no tests r
 | R3  |      No unique cap assignment      |            Assume there are >0 people and at least one cap per person. If there is no unique assignment of caps, the output should be 0. |
 | R4  | One or more unique cap assignments | Assume there are >0 people and at least one cap per person. If there is at least one unique assignment of caps, the output should be >0. |
 
+## Algorithm description
+To solve this problem, we will use bitmasking and dynamic programming. Bitmasking is used to represent subsets of a collection of elements as bits sequences called _masks_. In these sequences, a bit set to 1 means that the associated element is part of the subset. More specifically, if the _i-th_ bit is set to 1, then the _i-th_ element is part of the subset defined by this sequence. For example, if we have a collection of 10 elements, the bits sequence 0111010000 means that the associated subset contains element 2, 3, 4 and 6.
+
+In our case, bitmasking will be used to represent what the persons are wearing. In other words, a _i-th_ bit set to 1 means that the _i-th_ person is wearing a hat. Therefore the final cases we're interested into are the ones when the mask is completely full, i.e. all bits are set to 1.
```

Comment:
Shouldn't the first sentence read as "bitmasking will be used to represent _whether a person is wearing a hat_"?

Reply:
That is what I meant, the way I wrote it is confusing I agree, I'll update this